bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -1339,7 +1338,8 @@ public void writeComplete(int rc, long ledgerId, long entryId,
      * This method is idempotent. Once a ledger is fenced, it can
      * never be unfenced. Fencing a fenced ledger has no effect.
      */
-    public SettableFuture<Boolean> fenceLedger(long ledgerId, byte[] masterKey) throws IOException, BookieException {
+    public SettableFuture<Boolean> fenceLedger(long ledgerId, byte[] masterKey)
+            throws IOException, BookieException, InterruptedException {
```

Comment:
Same problem here

Reply:
Good catch, fixed. I left the original IOException and BookieException as they were. We can argue they should be on the future as well, though that's unrelated with this PR.